backend/tests/db.test.ts

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable jest/no-disabled-tests */
+/* eslint-disable sort-imports */
+
+import dotenv from 'dotenv';
+dotenv.config();
+
+import connectDB from '../src/db.ts';
+import mongoose from 'mongoose';
+
+// Mock the mongoose module
+jest.mock('mongoose', () => ({
+  connect: jest.fn(),
+}));
+
+// Mock console.log and console.error
+console.log = jest.fn();
+console.error = jest.fn();
+
+describe('connectDB', () => {
+  const originalEnv = process.env;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    process.env = { ...originalEnv };
+  });
+
+  afterAll(() => {
+    process.env = originalEnv;
+  });
+
+  it.skip('should connect to MongoDB Atlas in production environment', async () => {
+    process.env.NODE_ENV = 'production';
+    const atlasUri = process.env.ATLAS_URI;
+
+    await connectDB('testdb');
+
+    expect(mongoose.connect).toHaveBeenCalledWith(atlasUri);
+    expect(console.log).toHaveBeenCalledWith('Connected to MongoDB Atlas');
+  });
+
+  it('should connect to local MongoDB in development environment', async () => {
+    process.env.NODE_ENV = 'development';
+    const mongoUri = process.env.MONGO_URI;
+
+    await connectDB('testdb');
+
+    expect(mongoose.connect).toHaveBeenCalledWith(`${mongoUri}/testdb`);
+    expect(console.log).toHaveBeenCalledWith(`Connected to local MongoDB: ${mongoUri}/testdb`);
+  });
+
+  it('should connect to local test MongoDB in test environment', async () => {
+    process.env.NODE_ENV = 'test';
+
+    await connectDB('testdb');
+
+    expect(mongoose.connect).toHaveBeenCalledWith('mongodb://localhost:27017/testdb-test');
```

Comment:
Not a change request, but something to think about for the future is what the test environment will be. Will all unit tests only use the memory mongodb? Will the test environment have a dedicated test db, and will it always be in the same host as the test runner? Will everything be in a container and therefore have different host names?

Reply:
Great questions! For the record, we decided in @davidzlu [`utils`](#158 ) PR to have standalone memory mongodbs for each test suite that needs one. So we will not be using one database for the entire test environment which could lead to unexpected test results.